internal/compute/ansible.go

```diff
@@ -28,26 +49,123 @@ type AnsibleConfigurator struct {
 	sshKeyPath string
 	// mutex protects the instances map
 	mutex sync.Mutex
+	// instanceRepo provides access to instance data
+	instanceRepo InstanceRepository
+	// ownerID is the ID of the owner of the instances
+	ownerID uint
 }
 
 // NewAnsibleConfigurator creates a new Ansible configurator
-func NewAnsibleConfigurator(jobID string) *AnsibleConfigurator {
+func NewAnsibleConfigurator(jobID string, instanceRepo InstanceRepository) *AnsibleConfigurator {
 	return &AnsibleConfigurator{
-		jobID:     jobID,
-		instances: make(map[string]string),
+		jobID:        jobID,
+		instances:    make(map[string]string),
+		instanceRepo: instanceRepo,
 	}
 }
 
+// SetOwnerID sets the owner ID for the configurator
+func (a *AnsibleConfigurator) SetOwnerID(ownerID uint) {
```

Comment:
this is only called in the test right? I don't think it is needed outside of testing.
I would remove it and just update the ID in the test directly.